pkg/kubeapiserver/options/options.go

```diff
@@ -18,12 +18,27 @@ package options
 
 import (
 	"net"
+	"os"
 
 	utilnet "k8s.io/apimachinery/pkg/util/net"
+	genericoptions "k8s.io/apiserver/pkg/server/options"
 )
 
 // DefaultServiceNodePortRange is the default port range for NodePort services.
 var DefaultServiceNodePortRange = utilnet.PortRange{Base: 30000, Size: 2768}
 
 // DefaultServiceIPCIDR is a CIDR notation of IP range from which to allocate service cluster IPs
 var DefaultServiceIPCIDR net.IPNet = net.IPNet{IP: net.ParseIP("10.0.0.0"), Mask: net.CIDRMask(24, 32)}
+
+// DefaultProcessInfo is the default process information used for sending events
+func DefaultProcessInfo() *genericoptions.ProcessInfo {
+	// try and set name to the hostname if available
+	var name string
+	if name, _ = os.Hostname(); name == "" {
```

Comment:
our convention has been not to assign things that live past this block other than err.